code/clue.js

```diff
@@ -1,59 +1,350 @@
+};
 
 // CREATE OBJECTS FOR ALL THE WEAPONS, ADD MORE CHARACTERISTICS TO THE WEAPONS IF YOU LIKE.
 
 const rope = {
-  name: 'rope',
-  weight: 10
-}
+  name: "rope",
+  weight: 10,
+  id: "rope",
+};
+
+const knife = {
+  name: "knife",
+  weight: 5,
+  id: "knife",
+};
+
+const candlestick = {
+  name: "candlestick",
+  weight: 7,
+  id: "candlestick",
+};
+
+const dumbbell = {
+  name: "dumbbell",
+  weight: 30,
+  id: "dumbbell",
+};
+
+const poison = {
+  name: "poison",
+  weight: 2,
+  id: "poison",
+};
+
+const axe = {
+  name: "axe",
+  weight: 12,
+  id: "axe",
+};
+
+const bat = {
+  name: "bat",
+  weight: 13,
+  id: "bat",
+};
+
+const trophy = {
+  name: "trophy",
+  weight: 20,
+  id: "trophy",
+};
+
+const pistol = {
+  name: "pistol",
+  weight: 90,
+  id: "pistol",
+};
 
 // THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
 
+console.log(colonelMustard.color);
+console.log(pistol.weight);
+
 // NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  professorPlum,
+  missScarlet,
+  mrsPeackok,
+  colonelMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const weapons = []
+console.log(suspects);
+console.log(suspects[3].occupation);
+console.log(weapons);
+console.log(rooms.indexOf("Spa"));
 
-const rooms = []
+const weaponRooms = {
+  rope: ["Conservatory", "Hall", "Theater", "Patio", "Observatory"],
+  knife: ["Kitchen"],
+  candlestick: [
+    "Dining Room",
+    "Conservatory",
+    "Study",
+    "Library",
+    "Lounge",
+    "Ballroom",
+    "Hall",
+    "Spa",
+    "Living Room",
+    "Guest House",
+    "Patio",
+    "Theater",
+  ],
+  dumbbell: ["Guest House"],
+  poison: [
+    "Dining Room",
+    "Conservatory",
+    "Kitchen",
+    "Living Room",
+    "Guest House",
+    "Patio",
+  ],
+  axe: ["Conservatory", "Billiard Room", "Hall", "Patio"],
+  bat: ["Billiard Room", "Guest House"],
+  trophy: ["Living Room", "Guest House"],
+  pistol: ["Living Room", "Gest House"],
+};
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  killer: undefined,
+  weapon: undefined,
+  room: undefined,
+  favWeapon: undefined,
```

Comment:
I updated them to null, but as I remember on one of the lessons in similar scenario it was mentioned that values should be undefined.
Can you please share why its null better than undefined in this case?

Reply:
Sure Sandra! We use `undefined` values as initial ones when we create variables that we will want to update later on. Good example of that can be for example this piece of code, which will show user on website different alert, depending on user age :

```
let greeting; // Over here we initialise variable as undefined. NOTE ONE THING : we don't have to write let greeting = undefined, because let greeting; will do the same
if (userAge > 10) {
    greeting = 'Hey champ';
    alert(greeting);
} else if (userAge > 25) {
    greeting = 'Hello there';
    alert(greeting);
} else if (userAge > 50) {
    greeting = 'Good morning sir';
    alert(greeting);
}
```

Null however we should use mostly when we are talking about properties in object, exactly as in `mystery` over here. So we already know that `mystery` type is object, we just don't know yet exact values of properties